src/main/scala/firrtl/annotations/JsonProtocol.scala

```diff
@@ -277,6 +419,18 @@ object JsonProtocol {
           s"Annotations must be serialized as a JArray, got ${x.getClass.getName} instead!"
         )
     }
+
+    /* Tries to extract class name from the mapping exception */
+    def getAnnotationNameFromMappingException(mappingException: MappingException): String = {
+      try {
+        val GetClassPattern(name) = mappingException.getMessage
+        name
+      } catch {
+        case _: Exception =>
+          mappingException.getMessage
+      }
```

Comment:
```suggestion
      mappingException.getMessage match {
        case GetClassPattern(name) => name
        case other                                => other
      }
```
Simpler and more canonical style.

Reply:
Yes, of course.